src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -667,3 +668,20 @@ export const directToCorrectForm = ({
     router.push(returnUrl);
   }
 };
+
+/**
+ * Pushes an event to the Analytics dataLayer if the event doesn't already
+ * exist there.
+ * @param {object} event this will get pushed to `dataLayer`
+ * @param {string} key the property in the event object to use when looking for
+ *                     existing matches in the dataLayer
+ */
+export const recordEventOnce = (event, key) => {
+  const alreadyRecorded =
+    window.dataLayer &&
+    !!window.dataLayer.find(item => item[key] === event[key]);
```

Comment:
Probably worth putting in the comment that this function assumes `event` will contain only a single property. Pushing multiple properties into the `dataLayer` [is allowed](https://developers.google.com/tag-manager/devguide#multipush).